api/segments/models.py

```diff
@@ -45,14 +57,44 @@ class Segment(
         Feature, on_delete=models.CASCADE, related_name="segments", null=True
     )
 
+    # This defaults to 1 for newly created segments.
+    version = models.IntegerField(null=True)
+
+    # The related_name is not useful without specifying all_objects as a manager.
+    version_of = models.ForeignKey(
+        "self",
+        on_delete=models.CASCADE,
+        related_name="versioned_segments",
+        null=True,
+        blank=True,
+    )
     metadata = GenericRelation(Metadata)
 
+    # Only serves segments that are the canonical version.
+    objects = SegmentManager()
+
+    # Includes versioned segments.
+    all_objects = SoftDeleteExportableManager()
+
     class Meta:
         ordering = ("id",)  # explicit ordering to prevent pagination warnings
 
     def __str__(self):
         return "Segment - %s" % self.name
 
+    def get_skip_create_audit_log(self) -> bool:
+        skip = segment_audit_log_helper.should_skip_audit_log(self.id)
```

Comment:
After trying multiple different approaches including instance variables, `.initial_state`, `.state` all of which were wiped during the save process I finally resorted to using a helper to track the state of which audit logs can be created.